src/lib/Constants.sol

```diff
@@ -50,5 +50,27 @@ library Constants {
     address public constant BALANCER_VAULT = 0xBA12222222228d8Ba445958a75a0704d566BF2C8;
 
     // address of the 0x swap router contract
-    address public constant ZEROX_ROUTER = 0xDef1C0ded9bec7F1a1670819833240f027b25EfF;
+    address public constant ZERO_EX_ROUTER = 0xDef1C0ded9bec7F1a1670819833240f027b25EfF;
+
+    // address of the EULER protocol contract
+    address public constant EULER_PROTOCOL = 0x27182842E098f60e3D576794A5bFFb0777E025d3;
+    // address of the EULER markets contract
+    address public constant EULER_MARKETS = 0x3520d5a913427E6F0D6A83E07ccD4A4da316e4d3;
+    // address of the EULER eUSDC token contract (supply token)
+    address public constant EULER_EUSDC_TOKEN = 0xEb91861f8A4e1C12333F42DCE8fB0Ecdc28dA716;
+    // address of the EULER eWETH token contract (debt token)
+    address public constant EULER_DWETH_TOKEN = 0x62e28f054efc24b26A794F5C1249B6349454352C;
+    // address of the EULER rewards token contract
+    address public constant EULER_REWARDS_TOKEN = 0xd9Fcd98c322942075A5C3860693e9f4f03AAE07b;
+    // vaule used to scale the token's collateral/borrow factors from the euler market
+    uint32 constant EULER_CONFIG_FACTOR_SCALE = 4_000_000_000;
+
+    // address of the Aave v2 lending pool
+    address public constant AAVE_V2_LENDING_POOL = 0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9;
+    // address of the Aave v2 protocol data provider
+    address public constant AAVE_V2_PROTOCOL_DATA_PROVIDER = 0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d;
+    /// address of the Aave v2 interest bearing USDC (aUSDC) token
+    address public constant AAVE_V2_AUSDC_TOKEN = 0xBcca60bB61934080951369a648Fb03DF4F96263C;
+    // address of the Aave v2 variable debt bearing WETH (variableDebtWETH) token
+    address public constant AAVE_V2_VAR_DEBT_WETH_TOKEN = 0xF63B34710400CAd3e044cFfDcAb00a0f32E33eCf;
```

Comment:
put these in the individual adapters instead?

Reply:
[fixed](https://github.com/lindy-labs/sandclock-contracts/pull/58/commits/7b4c1270c508799eeff9b44cb4d4d30c023647b0)